PetrolEngine.cpp

```diff
@@ -10,11 +11,49 @@ PetrolEngine::PetrolEngine(int power, float capacity, int gears)
     std::cout << __FUNCTION__ << std::endl;
 }
 
+PetrolEngine::~PetrolEngine()
+{
+    std::cout << __FUNCTION__ << std::endl;
+}
+
 void PetrolEngine::changeGear(int gear)
 {
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+    if (isGearValid(gear))
+    {
+	try
+	{
+	    if (currentGear_ > 0 && gear == -1)
+	    {
+		throw InvalidGear();
+	    }
+	    else
+	    {
+		currentGear_ = gear;
+		std::cout << "Gear has been changed to: " << currentGear_ << std::endl;
+	    }
+
+	}
+	catch (const InvalidGear& e)
```

Comment:
znowu zle miejsce na catch

Reply:
uwazam ze dobrze, bo wczesniej sprawdzam czy bieg, na ktory chcemy zmienic miesci sie w przedziale <-1, 6> (defaultowy), a wyjatek chcialem rzucic tylko wtedy kiedy bieg znajduje sie w tym przedziale, ale podczas jazdy nie mozna wrzucic na R, czyli -1, zalozenie,. bieg na R mozna przerzucic tylko z biegu neutralnego.

Reply:
racja ale chodzilo mi o catcha a nie o throw ;)
jesli ja bede uzywal funkcje "changeGear" jak ja moge stwierdzic ze wszystko poszlo ok?

Reply:
Tak, tutaj raczej sam throw będzie lepszy, a catch powinien pójść tam, gdzie funkcja `changeGear()` jest wywoływana. Ale na razie nie zwracamy większej uwagi na takie rzeczy, to tylko sugestia. Wszystko w swoim czasie. Fajnie że użyliście tego wyjątku.